tough/src/error.rs

```diff
@@ -478,6 +496,62 @@ pub enum Error {
 
     #[snafu(display("Role missing from snapshot meta: {}", name))]
     RoleNotInMeta { name: String },
+
+    #[snafu(display("The key for {} was not included", role))]
+    KeyNotFound {
+        role: String,
+        source: schema::Error,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display("No keys were found for {}", role))]
+    NoKeys { role: String },
+
+    #[snafu(display("Invalid number"))]
+    InvalidInto {
+        source: std::num::TryFromIntError,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display("Invalid threshold number"))]
+    InvalidThreshold { backtrace: Backtrace },
+
+    /// The library failed to serialize an object to JSON.
+    #[snafu(display("Failed to serialize to JSON: {}", source))]
+    JsonSerialization {
+        source: schema::Error,
+        backtrace: Backtrace,
+    },
+
+    /// Invalid path permissions
+    #[snafu(display("Invalid path permission of {} : {:?}", name, paths))]
+    InvalidPathPermission {
+        name: String,
+        paths: Vec<String>,
+        source: schema::Error,
+    },
+
+    /// Overflow error with checked add
+    #[snafu(display("Overflow of checked add"))]
+    Overflow,
+
+    /// SignedDelegatedTargets has more than 1 signed targets
+    #[snafu(display("Not exactly 1 role was created: {}", count))]
```

Comment:
Let's clarify this message as well. Are we trying to say that more than 1 delegated targets roles was created?

Reply:
We are trying to say that 0 or more than 1 role was created. Maybe `Exactly 1 role was required, but {} were created`?

Reply:
That sounds better!